components/datetime/src/provider/gregory.rs

```diff
@@ -0,0 +1,263 @@
+                pub $element: $ty,
+            ));
+        };
+        ($name: ident { } -> ($($members: tt)*)) => {
+            pub mod $name {
+                use super::*;
+
+                #[derive(Debug, PartialEq, Clone, Default)]
+                #[cfg_attr(feature="provider_serde", derive(serde::Serialize, serde::Deserialize))]
+                pub struct SymbolsV1 {
+                    $($members)*
+                }
+                symbols!();
+            }
+        };
+        () => {
+            // UTS 35 specifies that `format` widths are mandatory
+            // except of `short`.
+            #[derive(Debug, PartialEq, Clone, Default)]
+            #[cfg_attr(feature="provider_serde", derive(serde::Serialize, serde::Deserialize))]
+            pub struct FormatWidthsV1 {
+                pub abbreviated: SymbolsV1,
+                pub narrow: SymbolsV1,
+                #[cfg_attr(
+                    all(feature="provider_serde", not(feature="serialize_none")),
+                    serde(skip_serializing_if = "Option::is_none"))
+                ]
+                pub short: Option<SymbolsV1>,
+                pub wide: SymbolsV1,
+            }
+
+            // UTS 35 specifies that `stand_alone` widths are optional
+            #[derive(Debug, PartialEq, Clone, Default)]
+            #[cfg_attr(feature="provider_serde", derive(serde::Serialize, serde::Deserialize))]
+            pub struct StandAloneWidthsV1 {
+                #[cfg_attr(
+                    all(feature="provider_serde", not(feature="serialize_none")),
+                    serde(skip_serializing_if = "Option::is_none"))
+                ]
+                pub abbreviated: Option<SymbolsV1>,
+                #[cfg_attr(
+                    all(feature="provider_serde", not(feature="serialize_none")),
+                    serde(skip_serializing_if = "Option::is_none"))
+                ]
+                pub narrow: Option<SymbolsV1>,
+                #[cfg_attr(
+                    all(feature="provider_serde", not(feature="serialize_none")),
+                    serde(skip_serializing_if = "Option::is_none"))
+                ]
+                pub short: Option<SymbolsV1>,
+                #[cfg_attr(
+                    all(feature="provider_serde", not(feature="serialize_none")),
+                    serde(skip_serializing_if = "Option::is_none"))
+                ]
+                pub wide: Option<SymbolsV1>,
+            }
+
+            #[derive(Debug, PartialEq, Clone, Default)]
+            #[cfg_attr(feature="provider_serde", derive(serde::Serialize, serde::Deserialize))]
+            pub struct ContextsV1 {
+                pub format: FormatWidthsV1,
+                #[cfg_attr(
+                    all(feature="provider_serde", not(feature="serialize_none")),
+                    serde(skip_serializing_if = "Option::is_none"))
+                ]
+                pub stand_alone: Option<StandAloneWidthsV1>,
```

Comment:
`question`: if all `StandAlongWidthsV1` are optional, is there a benefit to keeping `stand_alone` as an option as well?

Reply:
This might be a question for @sffc

I didn't actually edit or create this file. I just moved it form a local `mod gregory { ... }` to a file: `gregory.rs`.

Reply:
Yeah, I don't think this is actionable since it's pre-existing code.  The discussion on horizontal fallbacks (#259) should clarify how we deal with these things.